File: latex/main paper/python code/sample_euler_triangle.py
import numpy as np
def in_triangle(time, pos): return 1-time > abs(pos)
def sample_euler_triangle(dt=0.001):
    pos, time = 0, 0
    while in_triangle(time, pos):
        pos += np.random.normal(0, 1) * np.sqrt(dt)
        time += dt
    return (time, pos)

File: latex/main paper/julia_code/sample_euler_triangle.jl
in_triangle(time, pos) = (1 - time > abs(pos))
function sample_euler_triangle(dt=0.001)
    (pos = 0; time = 0)
    while in_triangle(time, pos)
        pos += sqrt(dt) * randn()
        time += dt
    end
    return (time, pos)
end